service/order.go

```diff
@@ -273,6 +276,44 @@ func (s *OrderService) Receive(buf []byte, from peer.ID) error {
 					err = errors.E(errors.Op("Put order"), err)
 				}
 			}
+		case pb.Operation_LOCK, pb.Operation_UNLOCK:
+			// Unmarshal order to get its key, validate
+			order := &pb.Order{}
+			err = proto.Unmarshal(data, order)
+			if !errors.IsEmpty(err) {
+				return errors.E(errors.Op("Unmarshal order proto in Receive"), err)
+			}
+
+			previousOrderData, err := s.Storage.Get(getOrderStorageKey(channelID, order.GetId()))
+			if !errors.IsEmpty(err) {
+				return errors.E(errors.Op("Get previous order"), err)
+			}
+			previousOrder := &pb.Order{}
+			proto.Unmarshal(previousOrderData, previousOrder)
+			if previousOrder.Nonce > order.Nonce {
+				return errors.E(errors.Op("Compare nonces"), "new order is older")
```

Comment:
errors.E(errors.Op("Compare nonces"), "received order state is behind current status")